File: sistema_municipal_frontend/src/app/services/like-dislike.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

export type VoteType = 'LIKE' | 'DISLIKE';

export interface ConteoVotos {
  likes: number;
  dislikes: number;
}

@Injectable({
  providedIn: 'root'
})
export class LikeDislikeService {
  private baseUrl = baseUrl;

  constructor(private http: HttpClient) {}

  votarForo(foroId: number, tipo: VoteType) {
    return this.http.post(`${this.baseUrl}/votos/foro/${foroId}?tipoVoto=${tipo}`, {});
  }

  votarComentario(comentarioId: number, tipo: VoteType) {
    return this.http.post(`${this.baseUrl}/votos/comentario/${comentarioId}?tipoVoto=${tipo}`, {});
  }

  // Métodos para obtener conteos
  obtenerConteosForo(foroId: number) {
    return this.http.get<ConteoVotos>(`${this.baseUrl}/votos/foro/${foroId}/conteos`);
  }

  obtenerConteosComentario(comentarioId: number) {
    return this.http.get<ConteoVotos>(`${this.baseUrl}/votos/comentario/${comentarioId}/conteos`);
  }
}

File: sistema_municipal_backend/src/main/java/com/utp/sistema_municipal_backend/service/LikeDislikeService.java
package com.utp.sistema_municipal_backend.service;

import com.utp.sistema_municipal_backend.model.*;

public interface LikeDislikeService {

    void votarForo(Long foroId, String emailUsuario, VoteType tipoVoto) throws Exception;

    void votarComentario(Long comentarioId, String emailUsuario, VoteType tipoVoto) throws Exception;

    long contarLikesForo(Long foroId);

    long contarDislikesForo(Long foroId);

    long contarLikesComentario(Long comentarioId);

    long contarDislikesComentario(Long comentarioId);
}
